lib/msal-common/src/request/AuthorizationCodeUrlParameters.ts

```diff
@@ -0,0 +1,146 @@
+ *                          login, none, and consent. The prompt=login claim forces the user to enter their credentials on that request,
+ *                          which negates single sign-on. The prompt=none claim is the opposite. This claim ensures that the user
+ *                          isn't presented with any interactive prompt at. If the request can't be completed silently via single
+ *                          sign-on, the Microsoft identity platform endpoint returns an error. The prompt=consent claim triggers
+ *                          the OAuth consent dialog after the user signs in. The dialog asks the user to grant permissions to the app.
+ *
+ * loginHint:               You can use this parameter to pre-fill the username and email address field of the sign-in page for
+ *                          the user, if you know the username ahead of time. Often, apps use this parameter during reauthentication,
+ *                          after already extracting the username from an earlier sign-in by using the preferred_username claim.
+ *
+ * domainHint:              The realm of the user in a federated directory. This skips the email-based discovery process
+ *                          that the user goes through on the sign-in page, for a slightly more streamlined user experience.
+ *                          For tenants that are federated through an on-premises directory like AD FS,
+ *                          this often results in a seamless sign-in because of the existing login session.
+ *
+ * claimsRequest:           facilitate the user to pass in "claims" requested the service for conditional access (CA),
+ *                          typically after CA fail
+ *
+ * nonce:                   A value included in the request, generated by the app, that will be included in the
+ *                          resulting id_token value as a claim. The app can verify this value to mitigate token replay attacks.
+ *                          The value typically is a randomized, unique string that can be used to identify the origin of the request.
+ *
+ * correlationId:           Unique GUID set per request to trace a request end-to-end for telemetry purposes
+ *
+ *
+ * This "Request" parameter is called when `msal-node` makes the authorization code request to the service on behalf of the app
+ */
+export class AuthorizationCodeUrlParameters {
+    scopes?: Array<string>;
+    authority?: string;
+    redirectUri?: string;
+    codeChallenge?: string;
+    codeChallengeMethod?: string;
+    state?: string;
+    prompt?: string;
+    loginHint?: string;
+    domainHint?: string;
+    claimsRequest?: string;
+    nonce?: string;
+    correlationId?: string;
+
+    /**
+     * This API validates the `AuthorizationCodeUrlParameters` and creates a URL to be sent to the server
+     * @param request
+     * @param config
+     */
+    static generateAuthCodeUrlParams(
```

Comment:
same comment about static functions and types.

Reply:
Agreed.

Reply:
done